packages/components/src/templates/next/components/internal/UnsupportedBrowserBanner/UnsupportedBrowserBanner.tsx

```diff
@@ -19,13 +20,32 @@ const getUserAgent = (): string | undefined => {
 const isSupportedBrowser = ({
   userAgent,
 }: UnsupportedBrowserBannerProps): boolean => {
+  if (userAgent === undefined) return true
+  if (userAgent === "") return true
   return supportedBrowsers.test(userAgent || getUserAgent() || "")
 }
 
 export const UnsupportedBrowserBanner = ({
-  userAgent,
+  userAgent: initialUserAgent,
 }: UnsupportedBrowserBannerProps) => {
-  if (isSupportedBrowser({ userAgent })) {
+  const [navigatorUserAgent, setNavigatorUserAgent] = useState(
+    initialUserAgent || "",
+  )
+
+  useEffect(() => {
+    // If no userAgent prop is provided, and we're on the client-side, set navigatorUserAgent from navigator
+    // The check for typeof window and navigator ensures this only runs in browser environments, not during server-side rendering
+    // We use setNavigatorUserAgent to update the state, which will trigger a re-render with the correct user agent
+    if (
+      !initialUserAgent &&
+      typeof window !== "undefined" &&
+      typeof navigator !== "undefined"
+    ) {
+      setNavigatorUserAgent(navigator.userAgent)
+    }
+  }, [initialUserAgent])
```

Comment:
i think since `navigator.userAgent` is a derived value, we might want to call `useMemo` rather htan `useEffect` here?

Reply:
if im not wrong, `useMemo` is for memoizing computed values to improve performance, not for running side effects or updating state (which we are doing here in this case). not sure if im missing out anything?

Reply:
just from looking, it looks like [this](https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state) might be what we're doing!

i'm not entirely sure here because of `window` and the fact that it might live outside of react but if we pass it to `useMemo` as a dep, it should function like a computed value

Reply:
@seaerchin `navigator.userAgent` is a derived value but the component wouldn't know it has changed since it's not dependent on that. 

This current approach is to indirectly force a re-render after the first render so that we can ideally retrieve the window.navigator.userAgent (if its available)

Reply:
unless you mean removing **both** `useState` and `useEffect` altogether and doing something like this?

<img width="590" alt="image" src="https://github.com/user-attachments/assets/c71f260b-4f71-48ba-9eca-2aaa0b547f89">

Reply:
yea i was thinking something like 
```ts
const navigatorUserAgent = useMemo(() => 
    initialUserAgent || 
    typeof window !== undefined && typeof navigator !== undefined && navigator.userAgent ||"", 
    [window, navigator, initialUserAgent]
)}
```

cos this would have the same rendering/re-rendering logic as the `useEffect + setState` above wtihout the extra logical hops